File: src/Moryx.Runtime.Maintenance/Maintenance/Logging/Models/LoggerModel.cs
// Copyright (c) 2020, Phoenix Contact GmbH & Co. KG
// Licensed under the Apache License, Version 2.0

using Moryx.Logging;

namespace Moryx.Runtime.Maintenance.Logging
{
    /// <summary>
    /// Model which represens a plugin logger. 
    /// </summary>
    public class LoggerModel
    {
        /// <summary>
        /// Name of this logger.
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// The level for which this logger is configured. See <see cref="LogLevel"/> for level information.
        /// </summary>
        public LogLevel ActiveLevel { get; set; }

        /// <summary>
        /// List of childs of this logger.
        /// </summary>
        public LoggerModel[] ChildLogger { get; set; }

        /// <summary>
        /// The parent of this logger. 
        /// </summary>
        public LoggerModel Parent { get; set; }
    }
}

File: src/Moryx.CommandCenter.Web/src/log/models/LoggerModel.ts
/*
 * Copyright (c) 2020, Phoenix Contact GmbH & Co. KG
 * Licensed under the Apache License, Version 2.0
*/
import { LogLevel } from "./LogLevel";

export default class LoggerModel {
    public name: string;
    public activeLevel: LogLevel;
    public childLogger: LoggerModel[];
    public parent: LoggerModel;

    public static shortLoggerName(logger: LoggerModel): string {
        const splittedLoggerPath = logger.name.split(".");
        return splittedLoggerPath[splittedLoggerPath.length - 1];
    }
}
